plugins/internal/telemetry/http_transport.go

```diff
@@ -0,0 +1,121 @@
+package telemetry
+
+import (
+	"io"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+// Refer OpenTelemetry Semantic Conventions for HTTP Client.
+// https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#http-client
+const (
+	metricClientDuration     = "http.client.duration"
+	metricClientRequestSize  = "http.client.request.size"
+	metricClientResponseSize = "http.client.response.size"
+
+	attributeServerPort         = "server.port"
+	attributeServerAddress      = "server.address"
+	attributeRequestMethod      = "http.request.method"
+	attributeResponseStatusCode = "http.response.status_code"
+)
+
+func NewHTTPTransport(baseTransport http.RoundTripper) http.RoundTripper {
+	if _, ok := baseTransport.(*httpTransport); ok {
+		return baseTransport
+	}
+
+	if baseTransport == nil {
+		baseTransport = http.DefaultTransport
+	}
+
+	icl := &httpTransport{roundTripper: baseTransport}
+	icl.createMeasures(otel.Meter("github.com/goto/meteor/plugins/internal/telemetry"))
+
+	return icl
+}
+
+type httpTransport struct {
+	roundTripper http.RoundTripper
+
+	counters       map[string]metric.Int64Counter
+	valueRecorders map[string]metric.Float64Histogram
+}
+
+func (tr *httpTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	ctx := req.Context()
+	startAt := time.Now()
+
+	var bw bodyWrapper
+	if req.Body != nil && req.Body != http.NoBody {
+		bw.ReadCloser = req.Body
+		req.Body = &bw
+	}
+
+	attribs := metric.WithAttributes(
+		attribute.String(attributeRequestMethod, req.Method),
+		attribute.String(attributeServerAddress, req.URL.Hostname()),
+		attribute.String(attributeServerPort, req.URL.Port()),
+	)
+
+	tr.counters[metricClientRequestSize].Add(ctx, bw.read.Load(), attribs)
+
+	resp, err := tr.roundTripper.RoundTrip(req)
+	elapsedTime := float64(time.Since(startAt)) / float64(time.Millisecond)
+	tr.valueRecorders[metricClientDuration].Record(ctx, elapsedTime, attribs)
```

Comment:
We should include the status code attribute here. Also, would it make sense to consistently include all attributes consistently for request size, response size and client duration?